intranet/apps/eighth/views/admin/scheduling.py

```diff
@@ -404,22 +406,44 @@ def transfer_students_action(request):
 
     num = source_act.members.count()
 
-    context = {"admin_page_title": "Transfer Students", "source_act": source_act, "dest_act": dest_act, "dest_unsignup": dest_unsignup, "num": num}
+    context = {
+        "admin_page_title": "Transfer Students",
+        "source_act": source_act,
+        "dest_act": dest_act,
+        "dest_unsignup": dest_unsignup,
+        "num": num,
+        "moved_students": None,
+    }
 
     if request.method == "POST":
         if dest_unsignup and not dest_act:
             source_act.eighthsignup_set.all().delete()
             invalidate_obj(source_act)
             messages.success(request, "Successfully removed signups for {} students.".format(num))
         else:
-            source_act.eighthsignup_set.update(scheduled_activity=dest_act)
+            duplicate_sign_up_users = []
+            duplicate_signups = []
+            with transaction.atomic():
+                for u in source_act.members.all():
+                    conflict_signup = u.eighthsignup_set.filter(scheduled_activity__block=dest_act.block)
+                    if conflict_signup.exists():
+                        duplicate_sign_up_users.append(u)
+                        duplicate_signups.append(conflict_signup[0])
+                lock_on(duplicate_sign_up_users)
+                for signup in duplicate_signups:
+                    signup.delete()
+
+                source_act.eighthsignup_set.update(scheduled_activity=dest_act)
+
+            transferred_activity_email.delay(dest_act, duplicate_sign_up_users)
+
+            context["moved_students"] = duplicate_signups
 
             invalidate_obj(source_act)
             invalidate_obj(dest_act)
             messages.success(request, "Successfully transfered {} students.".format(num))
-        return redirect("eighth_admin_dashboard")
```

Comment:
I would keep the redirect if there are no duplicates, but render a different "transfer_students_results" page if there are any duplicates.

Reply:
I think it is useful to have the transfer messages appear beneath the original text and this stays in theme with many of the 8th admin functions, which all have the appearance of occurring on the same page. Redirecting to another page would also involve repeating a lot of code that is already in this view. I'd rather just redirect to "eighth_admin_dashboard" if there are no duplicates.

Reply:
My concern is that such a "return the same-page" may cause an admin to press the "Transfer" button again; although that won't be a concern (as pressing the "Transfer" button again doesn't appear to break anything), but I think that it'd be better off to redirect to somewhere without a "Transfer" button.

Reply:
Okay the latest rendition (just pushed) keeps it on the same page as far as django is concerned, but hides the "Transfer" section with template tags. This mimics the behavior of the rest of the page, which uses template tags to show/hide the wizard.

Reply:
I'm happy with that. @etnguyen03 ?

Reply:
Good to me as well.